Jenkinsfile

```diff
@@ -0,0 +1,78 @@
+library 'magic-butler-catalogue'
+def PROJECT_NAME = 'logdna-python'
+def TRIGGER_PATTERN = ".*@logdnabot.*"
+
+pipeline {
+  agent none
```

Comment:
does this mean the job will run on Jenkins main node, rather than a worker? if so, what is the benefit of that vs running on a worker?

Reply:
It disabled the global "agent" for the pipeline and forces each stage to declare it's own, it'll still run on a worker node

Reply:
You could add an explicit label to each stage's agent block if you wanted to double enforce it

Reply:
I think it just means the stages need to be explicit about it

```txl
Defining agent none at the top-level of the Pipeline ensures that an Executor will not be assigned unnecessarily. Using agent none also forces each stage section to contain its own agent section
```